snmp/datadog_checks/snmp/discovery.py

```diff
@@ -45,7 +45,7 @@ def discover_instances(config, interval, check_ref):
             try:
                 profile = check._profile_for_sysobject_oid(sys_object_oid)
             except ConfigurationError:
-                if not (host_config.scalar_oids or host_config.bulk_oids):
+                if not (host_config.oid_config.scalar_oids or host_config.oid_config.bulk_oids):
```

Comment:
I _think_ the absence of `host_config.next_oids` here was originally a mistake, and we can update this to:

```suggestion
                if not host_config.oid_config.has_oids():
```

Reply:
Yes, that's likely the case. cc @therve